edb/server/compiler/compiler.py

```diff
@@ -1917,7 +1917,7 @@ def _compile_ql_query(
     sql_text = pg_codegen.generate_source(sql_ast)
     # If requested, embed the EdgeQL text in the SQL.
     if debug.flags.edgeql_text_in_sql and source:
-        sql_text += '\n-- ' + repr(source.text())
+        sql_text += '\n-- ' + json.dumps(source.text())
```

Comment:
Can we also prepend instead of appending it?  Postgres truncates large queries in its diagnostic views, so this will almost always be truncated away.